crates/driver/src/infra/blockchain/token.rs

```diff
@@ -111,29 +116,47 @@ impl Erc20 {
         source: order::SellTokenBalance,
         interactions: &[eth::Interaction],
     ) -> Result<eth::TokenAmount, Error> {
+        let mut method = self.balances.balance(
+            (
+                self.balances.address(),
+                self.vault_relayer.into(),
+                self.vault.into(),
+            ),
+            trader.into(),
+            self.token.address(),
+            0.into(),
+            ethcontract::Bytes(source.hash().0),
+            interactions
+                .iter()
+                .map(|i| {
+                    (
+                        i.target.into(),
+                        i.value.into(),
+                        ethcontract::Bytes(i.call_data.0.clone()),
+                    )
+                })
+                .collect(),
+        );
+        let access_list_call = contracts::storage_accessible::call(
+            method.tx.to.unwrap(),
+            contracts::bytecode!(contracts::support::Balances),
+            method.tx.data.clone().unwrap(),
+        );
+        // Create the access list for the balance simulation
+        if let Some(from) = &access_list_call.from {
```

Comment:
Unless I'm missing something this is not necessary. Either from is not set at all or it's using the zero address. Either case could be used to simplify this code.